icd/api/pipeline_compiler.cpp

```diff
@@ -1047,25 +1047,90 @@ void PipelineCompiler::GetPipelineCreationInfoNext(
 // =====================================================================================================================
 VkResult PipelineCompiler::SetPipelineCreationFeedbackInfo(
     const VkPipelineCreationFeedbackCreateInfoEXT* pPipelineCreationFeadbackCreateInfo,
-    const PipelineCreationFeedback*                pPipelineFeedback)
+    const VkGraphicsPipelineCreateInfo*            pCreateInfo,
+    const PipelineCreationFeedback*                pPipelineFeedback,
+    const PipelineCreationFeedback*                pStageFeedback)
 {
-    if (pPipelineCreationFeadbackCreateInfo != nullptr)
-    {
-        if (pPipelineFeedback->feedbackValid)
+  if (pPipelineCreationFeadbackCreateInfo != nullptr)
+  {
+    if (pPipelineFeedback->feedbackValid)
+    {
+      pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
+          VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+    }
+    if (pPipelineFeedback->hitApplicationCache)
+    {
+      pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
+          VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
+    }
+    pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->duration =
+        pPipelineFeedback->duration;
+    if (pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks !=
+            nullptr &&
+        pStageFeedback != nullptr)
+    {
+      // Compute pipeline does have VkGraphicsPipelineCreateInfo.
+      if (pCreateInfo == nullptr)
+      {
+        pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks[0]
```

Comment:
Maybe create a reference/pointer to this array element to avoid spelling out the full access path any time you want to update it?

Reply:
done